examples/notebooks/get-started/learn_beam_transforms_by_doing.ipynb

```diff
@@ -0,0 +1,740 @@
+      },
+      "outputs": [],
+      "source": [
+        "# Run a shell command and import beam\n",
+        "!pip install --quiet apache-beam\n",
+        "import apache_beam as beam\n",
+        "beam.__version__"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Set the logging level to reduce verbose information\n",
+        "import logging\n",
+        "\n",
+        "logging.root.setLevel(logging.ERROR)"
+      ],
+      "metadata": {
+        "id": "vyksB2VMtv3m"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "\n",
+        "\n",
+        "---\n",
+        "\n",
+        "\n",
+        "\n",
+        "---\n",
+        "\n"
+      ],
+      "metadata": {
+        "id": "M1ku4nX_Gutb"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## 1. Simple User-Defined Function (UDF)\n",
+        "\n",
+        "Some `PTransforms` allow you to run your own functions and user-defined code to specify how your transform is applied. For example, the below `CombineGlobally` transform,"
+      ],
+      "metadata": {
+        "id": "0qDeT34SS1_8"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "pc = [1, 10, 100, 1000]\n",
+        "\n",
+        "# User-defined function\n",
+        "def bounded_sum(values, bound=500):\n",
+        "  return min(sum(values), bound)\n",
+        "\n",
+        "small_sum = pc | beam.CombineGlobally(bounded_sum)  # [500]\n",
+        "large_sum = pc | beam.CombineGlobally(bounded_sum, bound=5000)  # [1111]\n",
+        "\n",
+        "print(small_sum, large_sum)"
+      ],
+      "metadata": {
+        "id": "UZTWBGZ0TQWF"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## 2. Transforms: ParDo and Combine\n",
+        "\n",
+        "A `ParDo` transform considers each element in the input `PCollection`, performs your user code to process each element, and emits zero, one, or multiple elements to an output `PCollection`. `Combine` is another Beam transform for combining collections of elements or values in your data.\n",
+        "Both allow flexible UDF to define how you process the data."
+      ],
+      "metadata": {
+        "id": "UBFRcPO06xiV"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### 2.1 DoFn\n",
+        "\n",
+        "DoFn - a Beam Python class that defines a distributed processing function (used in [ParDo](https://beam.apache.org/documentation/programming-guide/#pardo))"
+      ],
+      "metadata": {
+        "id": "P4W-1HIiV-HP"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "data = [1, 2, 3, 4]\n",
+        "\n",
+        "# create a DoFn to multiply each element by five\n",
+        "# you can define the procesing code under `process`\n",
```

Comment:
We should make clear that having a process function is a requirement for a DoFn.

Reply:
Added this to the comments. Thanks.